src/handlers/tests/status-retriever-handler.test.ts

```diff
@@ -0,0 +1,251 @@
+};
+
+const nullUpdatedAt = {
+  pk: 'testUserID',
+  intervention: 'some intervention',
+  updatedAt: null,
+  appliedAt: 12345685809,
+  sentAt: 123456789,
+  reprovedIdentityAt: 849473,
+  resetPasswordAt: 5847392,
+  blocked: false,
+  suspended: true,
+  resetPassword: false,
+  reproveIdentity: false,
+  auditLevel: 'standard',
+  ttl: 1234567890,
+  history: ['some intervention'],
+};
+
+const suspendedAccount = {
+  updatedAt: 123455,
+  appliedAt: 12345685809,
+  sentAt: 123456789,
+  description: suspendedRecord.intervention,
+  reprovedIdentityAt: 849473,
+  resetPasswordAt: 5847392,
+  state: {
+    blocked: false,
+    suspended: true,
+    resetPassword: false,
+    reproveIdentity: false,
+  },
+  auditLevel: 'standard',
+};
+
+const updatedTime = {
+  updatedAt: 1685404800000,
+  appliedAt: 12345685809,
+  sentAt: 123456789,
+  description: suspendedRecord.intervention,
+  reprovedIdentityAt: 849473,
+  resetPasswordAt: 5847392,
+  state: {
+    blocked: false,
+    suspended: true,
+    resetPassword: false,
+    reproveIdentity: false,
+  },
+  auditLevel: 'standard',
+};
+
+const mockConfig = ContextExamples.helloworldContext;
+const mockDynamoDBServiceRetrieveRecords = DynamoDbService.prototype.retrieveRecordsByUserId as jest.Mock;
+
+describe('status-retriever-handler', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
+    jest.setSystemTime(new Date(Date.UTC(2023, 4, 30)));
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+
+  it('will return the correct response from the database if the user ID matches', async () => {
+    mockDynamoDBServiceRetrieveRecords(testEvent.pathParameters ? ['userId'] : 'testUserID');
+    mockDynamoDBServiceRetrieveRecords.mockResolvedValueOnce([marshall(suspendedRecord)]);
+    const response = await handle(testEvent, mockConfig);
+    expect(response.statusCode).toBe(200);
+    expect(response.body).toEqual(JSON.stringify({ intervention: suspendedAccount }));
+  });
+
+  it('will return a message if user ID cannot be found in the database', async () => {
+    const response = await handle(testEvent, mockConfig);
+    expect(response.statusCode).toBe(200);
+    expect(response.body).toEqual(JSON.stringify({ intervention: AISInterventionTypes.AIS_NO_INTERVENTION }));
+  });
+
+  it('will return the correct response from the database if the user ID matches an account where the state items are all false', async () => {
+    mockDynamoDBServiceRetrieveRecords(testEvent.pathParameters ? ['userId'] : 'testUserID');
+    mockDynamoDBServiceRetrieveRecords.mockResolvedValueOnce([marshall(accountFoundNotSuspendedRecord)]);
+    const response = await handle(testEvent, mockConfig);
+    expect(response.statusCode).toBe(200);
+    expect(response.body).toEqual(JSON.stringify({ intervention: AISInterventionTypes.AIS_NO_INTERVENTION }));
+  });
+
+  it('will return the appropiate response if the event is malformed', async () => {
```

Comment:
This test is a bit vague. The description should be something more like: `will return status 400 if the userId in the event is empty`. I think it would also be easier to understand that this is what the test is actually testing if `invalidTestEvent` was created in this it block by doing something like this:
```
const invalidPathParameters = pathParameters: {
    proxy: '/ais/{user_id}',
    userId: ' ',
  }
const invalidTestEvent = { ...testEvent, pathParameters: invalidPathParameters }
```